### PR TITLE
Support monitoring for deployed Omen agents

### PR DESCRIPTION
**Pull request by evangriffiths**
Screenshot 2024-02-23 at 16 57 29

## Summary by CodeRabbit

- **New Features**
	- Enhanced prediction market tooling with dynamic agent loading based on market type.
- **Improvements**
	- Updated logic for determining boolean outcomes in prediction markets.
	- Streamlined bet retrieval method names for clarity.
- **Refactor**
	- Generalized agent loading mechanism to improve flexibility and maintenance.

### CODE REVIEW

File: prediction_market_agent_tooling/markets/omen/omen.py [hunk 1]
```diff
@@ -705,7 +705,7 @@ def to_int_timestamp(dt: datetime) -> int:
     return int(dt.timestamp())
 
 
-def get_resolved_bets(
+def get_bets(
```
Comment (evangriffiths) [inline prediction_market_agent_tooling/markets/omen/omen.py:708]: Renamed, as this method doesn't actually filter by resolved bets

File: prediction_market_agent_tooling/markets/omen/data_models.py [hunk 2]
```diff
@@ -120,7 +120,8 @@ def boolean_outcome(self) -> bool:
         if not self.is_resolved:
             raise ValueError(f"Bet with title {self.title} is not resolved.")
 
-        return get_boolean_outcome(check_not_none(self.currentAnswer))
+        outcome: str = self.outcomes[int(check_not_none(self.currentAnswer), 16)]
```
Comment (kongzii) [inline prediction_market_agent_tooling/markets/omen/data_models.py:123]: Why 16?

**Review by kongzii — APPROVED** [review]

File: prediction_market_agent_tooling/markets/omen/data_models.py (hunk 2 above)
Comment (evangriffiths) [reply prediction_market_agent_tooling/markets/omen/data_models.py:123]: The `currentAnswer` is in hex form (base-16).
```
>>> int("0xF", 16)
15
```

**Pull request merged**